src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+    }
+
+    if (ConfGetChildValueBoolWithDefault(if_root, if_default, dpdk_yaml.multicast, &entry_bool) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.multicast);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetMulticast(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks, &entry_bool) != 1) {
+        SCLogError(
+                SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.checksum_checks);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumChecks(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks_offload, &entry_bool) != 1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing",
+                dpdk_yaml.checksum_checks_offload);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumOffload(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_mode, &copy_mode_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_mode);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_iface, &copy_iface_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_iface);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetCopyIfaceSettings(iconf, copy_iface_str, copy_mode_str);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (iconf->copy_mode == DPDK_COPY_MODE_NONE) {
+        SCLogInfo("IDS mode enabled, disabling TX settings");
+        iconf->nb_tx_queues = 0;
+        iconf->nb_tx_desc = 0;
+    }
+
+    SCReturnInt(0);
+}
+DPDKIfaceConfig *ConfigParse(const char *iface)
+{
+    int retval;
+    DPDKIfaceConfig *iconf = NULL;
+    if (iface == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "Provided iface is NULL");
+        SCReturnPtr(NULL, "void *");
+    }
+
+    ConfigInit(&iconf);
+    if (iconf == NULL) {
+        SCReturnPtr(NULL, "void *");
+    }
+    retval = ConfigLoad(iconf, iface);
+    if (retval < 0) {
+        iconf->DerefFunc(iconf);
+        SCReturnPtr(NULL, "void *");
+    }
+
+    return iconf;
+}
+
+void DeviceInitPortConf(const DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        struct rte_eth_conf *port_conf)
+{
+    *port_conf = (struct rte_eth_conf){
+            .rxmode = {
+                    .mq_mode = ETH_MQ_RX_NONE,
+                    .max_rx_pkt_len = RTE_ETHER_MAX_LEN,
```

Comment:
the value is `#define RTE_ETHER_MAX_LEN   1518`. in earlier config code you were checking for jumbo frames and all. I do not see any check in port_init to enable JUMBO frame sizes also.

Reply:
Reworked, now I set it to the MTU size. Here I am checking also if Jumbo frames are supported by the NIC.